engine/src/main/java/com/datatorrent/stram/LaunchContainerRunnable.java

```diff
@@ -230,6 +230,12 @@ public void run()
       }
     }
 
+    String loggerAppender = dag.getValue(LogicalPlan.LOGGER_APPENDER);
+    if (loggerAppender != null) {
+      vargs.add(String.format("-D%s=\"%s\"", LogicalPlan.LOGGER_APPENDER.getName(), loggerAppender));
+      vargs.add(String.format("-D%s=\"%s\"", LogicalPlan.APPLICATION_NAME.getName(), dag.getValue(LogicalPlan.APPLICATION_NAME)));
```

Comment:
Application name should be always set. I may be good to have multiple properties set (like application name, container id and maybe more). Please file a separate JIRA (and PR).